File: easy/1399. Count Largest Group/1399. Count Largest Group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        nums = {}
        max_size = 0
        res = 0

        for num in range(1, n + 1):
            d_sum = sum(map(int, str(num)))

            if d_sum in nums:
                nums[d_sum].append(num)
            else:
                nums[d_sum] = [num]

            s = len(nums[d_sum])

            if s == max_size:
                res += 1
            elif s > max_size:
                res = 0
                max_size = s

        return res + 1

File: easy/1399. Count Largest Group/1399. Count Largest Group.ts
function countLargestGroup(n: number): number {
    const nums: { [key in number]: number[] } = {}
    let res = 0
    let maxSum = 0

    for (let num = 1; num <= n; num++) {
        const dSum = num > 9 ? num.toString().split("").reduce((acc, curr) => acc + Number(curr), 0) : num

        if (dSum in nums)
            nums[dSum].push(num)
        else
            nums[dSum] = [num]

        const s = nums[dSum].length
        if (s > maxSum) {
            res = 0
            maxSum = s
        } else if (s == maxSum) {
            res += 1
        }
    }

    return res + 1
};